onnxscript/ir/passes/common/shape_inference.py

```diff
@@ -22,12 +22,9 @@
 _BIG_TENSOR_SIZE_LIMIT = 1000  # 1KB
 
 
-class ShapeInferencePass(ir.passes.PassBase):
+class ShapeInferencePass(ir.passes.OutOfPlacePass):
```

Comment:
Just wondering aloud ... we are forced to make this OutOfPlace because of the existing API (to ONNX's shape inference) ... but that does come with some disadvantages ... eg., if we ever store extra information in the IR (that doesn't get serialized and is lost) ... even if some user/pass uses a dictionary mapping values/nodes in IR to some extra information ... should we, in the long run, prefer in-place? We could potentially do it (at least in the long run) ...

Reply:
I think being out of place is ok. The issues arise when we serialize. In the long run if we don't need to serialize then the info are more easily preserved, if we copy the objects properly.